model/textparse/openmetricsparse.go

```diff
@@ -222,9 +225,108 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	var lbs labels.Labels
+	p.Metric(&lbs)
+	lbs = lbs.DropMetricName()
+
+	newParser := deepCopyParser(p)
+loop:
+	for {
+		switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			// TODO: potentially broken? Missing type?
+			if newParser.mName != p.mName {
+				return nil
+			}
+
+			var newLbs labels.Labels
+			newParser.Metric(&newLbs)
+			name := newLbs.Get(model.MetricNameLabel)
+			if !strings.HasSuffix(name, "_created") {
+				continue
+			}
+
+			// edge case: if gauge_created of unknown type -> skip parsing
+			newLbs = newLbs.DropMetricName()
+			switch p.mtype {
+			case model.MetricTypeCounter:
+				if !labels.Equal(lbs, newLbs) {
+					return nil
+				}
+			case model.MetricTypeSummary:
+				labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
+				if labelDiffs.Len() != 0 {
+					if !labelDiffs.Contains("quantile") || labelDiffs.Len() != 1 {
+						return nil
+					}
+				}
+			case model.MetricTypeHistogram:
+				labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
+				if labelDiffs.Len() != 0 {
+					if !labelDiffs.Contains("le") || labelDiffs.Len() != 1 {
+						return nil
+					}
+				}
+			default:
+				// if mtype is not counter, summary or histogram, it won't have a valid _created line
+				return nil
+			}
+
+			ct := int64(newParser.val)
+			return &ct
+		default:
+			break loop
+		}
+	}
 	return nil
 }
 
+// We need this because we want offsets of the original parser unchanged when
+// we're working with a new parser in CreatedTimeStamp().
+func deepCopyParser(p *OpenMetricsParser) OpenMetricsParser {
+	newB := make([]byte, len(p.l.b))
+	copy(newB, p.l.b)
+
+	newLexer := &openMetricsLexer{
+		b:     newB,
+		i:     p.l.i,
+		start: p.l.start,
+		err:   p.l.err,
+		state: p.l.state,
+	}
+
+	newSeries := make([]byte, len(p.series))
+	copy(newSeries, p.series)
+	newText := make([]byte, len(p.text))
+	copy(newText, p.text)
+	newOffsets := make([]int, len(p.offsets))
+	copy(newOffsets, p.offsets)
+	newEOffsets := p.eOffsets
+	newExemplar := p.exemplar
+	newParser := OpenMetricsParser{
+		l:       newLexer,
+		builder: p.builder,
+		series:  newSeries,
+		text:    newText,
+		mtype:   p.mtype,
+		mName:   p.mName,
+		val:     p.val,
+		ts:      p.ts,
+		hasTS:   p.hasTS,
+		start:   p.start,
+
+		offsets: newOffsets,
+
+		eOffsets:      newEOffsets,
+		exemplar:      newExemplar,
+		exemplarVal:   p.exemplarVal,
+		exemplarTs:    p.exemplarTs,
+		hasExemplarTs: p.hasExemplarTs,
+		skipCT:        false,
+	}
```

Comment:
```suggestion
	newOffsets := make([]int, len(p.offsets))
	copy(newOffsets, p.offsets)
	newParser := OpenMetricsParser{
		l:       newLexer,
		start:   p.start,
		offsets: newOffsets,
		skipCT:        false,
	}
```

We use very little information from the old parser, right? Do we really need to copy everything? I'm not sure if my suggestion here is 100% correct, but you could try removing fields one-by-one while verifying that your tests still pass after each removal

Reply:
If we're not doing an entire copy, we could also rename this function from `deepCopyParser` to something more aligned with what we're doing, like `newParserForCreatedTimestamp` or `prepareParserForCreatedTimestamp`... whatever sounds better to you :)

Reply:
Kept the entire lexer for now and removed everything I could without breaking current tests and called it `createdTimestampParser`